### PR TITLE
Syndication Fix + JSON Format Support

### PR DESCRIPTION
**Pull request by effone**
Outdated PR. Refer: #3682

### CODE REVIEW

**Comment by Eldenroot** [comment]
I would like to see at least point 2 in the list above for MyBB 1.8.x branch. What do you think?

**Comment by effone** [comment]
I have tested further, since the restricted forums are no way accessible by readers I think we can live with this. The concern what bothered me is if you are logged in the feed shows as per your permission, its good actually - coz it becomes dynamic to produce feed to you respecting your permission to the site the feed is being fetched from.

I am differing the point.

**Comment by Eldenroot** [comment]
I have been testing this since yesterday. Personally I am not a fan of RSS and I do not use JSON format. Some users on my forum use it, I contacted them to test this feature and let me know. So far so good.

File: install/resources/mybb_theme.xml [hunk 1]
```diff
@@ -6192,7 +6192,7 @@ if(use_xmlhttprequest == "1")
 <td class="{$class}">{$smilie['name']}</td>
 <td class="{$class}">{$smilie['find']}</td>
 </tr>]]></template>
-		<template name="misc_syndication" version="120"><![CDATA[<html>
+		<template name="misc_syndication" version="1820"><![CDATA[<html>
```
Comment (euantorano) [inline install/resources/mybb_theme.xml]: This should be `1821`.

File: inc/class_feedgeneration.php [hunk 2]
```diff
@@ -121,17 +135,42 @@ function generate_feed()
 			}
 			switch($this->feed_format)
 			{
-				// Output Atom 1.0 format feed.
+				// Output JSON formatted feed.
+				case "json":
+					++$serial;
+					$end = $serial < count($this->items) ? "," : "";
+					$item_id = explode('tid=', $item['link']);
```
Comment (euantorano) [inline inc/class_feedgeneration.php]: Wouldn't it be easier to just build an army for the entry, then `json_encode` the whole array? eg:

```php
$entry = array(
	'url' => $item['link'],
	'title' => $item['title'],
	...
);

if(!empty($item['author']))
{
	$entry['author'] = array(
		...
	);
}

$this->feed .= json_encode($entry);
$this->feed .= $end . "\n";
```

The same would apply to the feed header. If nothing else, it's probably easier to read with an array in my opinion.

Comment (effone) [reply inc/class_feedgeneration.php]: I understand what you say. The reason doing it that way is to maintain the indentation in the similar manner and inline with how it has been done already for xml feeds.
If you don't bother about manual beautifying and insist I can do the suggested change. Please confirm.

**Comment by euantorano** [comment]
I’m not really bothered either way to be honest - both ways work after all.

**Comment by effone** [comment]
Superseded by #3682

**Pull request closed without merging**